scripts/Muon/GUI/Common/pairing_table_widget/pairing_table_widget_view.py

```diff
@@ -0,0 +1,345 @@
+        self.vertical_layout.addLayout(self.horizontal_layout)
+
+        self.setLayout(self.vertical_layout)
+
+    def set_up_table(self):
+        self.pairing_table.setColumnCount(5)
+        self.pairing_table.setHorizontalHeaderLabels(["Pair Name", "Group 1", " Group 2", "Alpha", "Guess Alpha"])
+        header = self.pairing_table.horizontalHeader()
+        header.setResizeMode(0, QtGui.QHeaderView.Stretch)
+        header.setResizeMode(1, QtGui.QHeaderView.Stretch)
+        header.setResizeMode(2, QtGui.QHeaderView.Stretch)
+        header.setResizeMode(3, QtGui.QHeaderView.Stretch)
+        header.setResizeMode(4, QtGui.QHeaderView.ResizeToContents)
+        vertical_headers = self.pairing_table.verticalHeader()
+        vertical_headers.setMovable(False)
+        vertical_headers.setResizeMode(QtGui.QHeaderView.ResizeToContents)
+        vertical_headers.setVisible(True)
+
+        self.pairing_table.horizontalHeaderItem(0).setToolTip("The name of the pair :"
+                                                              "\n    - The name must be unique across all groups/pairs"
+                                                              "\n    - The name can only use digits, characters and _")
+        self.pairing_table.horizontalHeaderItem(1).setToolTip("Group 1 of the pair, selected from the grouping table")
+        self.pairing_table.horizontalHeaderItem(2).setToolTip("Group 2 of the pair, selected from the grouping table")
+        self.pairing_table.horizontalHeaderItem(3).setToolTip("The value of Alpha for the pair asymmetry:"
+                                                              "\n   - The number must be >= 0.0")
+        self.pairing_table.horizontalHeaderItem(4).setToolTip("Replace the current value of Alpha with one estimated"
+                                                              " from the data.")
+
+    def num_rows(self):
+        return self.pairing_table.rowCount()
+
+    def num_cols(self):
+        return self.pairing_table.columnCount()
+
+    def update_group_selections(self, group_name_list):
+        self._group_selections = group_name_list
+
+    def get_index_of_text(self, selector, text):
+        for i in range(selector.count()):
+            if str(selector.itemText(i)) == text:
+                return i
+        return 0
+
+    def clear(self):
+        # Go backwards to preserve indices
+        for row in reversed(range(self.num_rows())):
+            self.pairing_table.removeRow(row)
+
+    def notify_data_changed(self):
+        if not self._updating:
+            self.dataChanged.emit()
+
+    def add_entry_to_table(self, row_entries):
+        assert len(row_entries) == self.pairing_table.columnCount() - 1
+
+        row_position = self.pairing_table.rowCount()
+        self.pairing_table.insertRow(row_position)
+        for i, entry in enumerate(row_entries):
+            item = QtGui.QTableWidgetItem(entry)
+            if i == 0:
```

Comment:
can we do something similar to the group table view where these are not magic numbers?